contracts/FlowIDTableStaking.cdc

```diff
@@ -768,10 +768,16 @@ pub contract FlowIDTableStaking {
         /// Sets a list of approved node IDs for the next epoch
         /// Nodes not on this list will be unstaked at the end of the staking auction
         /// and not considered to be a proposed/staked node
-        pub fun setApprovedList(_ nodeIDs: [String]) {
-            let list = FlowIDTableStaking.account.load<[String]>(from: /storage/idTableApproveList)
+        pub fun setApprovedList(_ nodeIDs: {String: Bool}) {
```

Comment:
Contract upgrades don't allow adding fields to composite types, so the lists that we've added and are adding have to be stored in account storage instead of a field. Yes, the account owner can change it, but that is currently gated by the service account multisig, so no transactions would be able to be run without the service account committee approving it.

Reply:
That makes sense. Maybe it will be better to have a code comment on top of the contract about the design choice because by seeing this contract, other community members can follow the same trend, and they will miss the key security aspect (Revoked the key of the contract account) to work with this design.